cypress/support/commands.ts

```diff
@@ -55,3 +61,42 @@ Cypress.Commands.add('generateEmptyApplication', () => {
     importedFromLegacyDatabase: false,
   });
 });
+
+Cypress.Commands.add('loginAsUser', (userType: string) => {
+  // if it's running against the local env then should be set from the env vaars, otherwise match the congg.
+  const users = {
+    officer: {
+      email: faker.internet.email({ provider: 'hackneyTEST.gov.uk' }),
+      name: faker.person.fullName(),
+      groups: Cypress.env('AUTHORISED_OFFICER_GROUP'),
+    },
+    manager: {
+      email: faker.internet.email({ provider: 'hackneyTEST.gov.uk' }),
+      name: faker.person.fullName(),
+      groups: Cypress.env('AUTHORISED_MANAGER_GROUP'),
+    },
+    admin: {
+      email: faker.internet.email({ provider: 'hackneyTEST.gov.uk' }),
+      name: faker.person.fullName(),
+      groups: Cypress.env('AUTHORISED_ADMIN_GROUP'),
+    },
+    readOnly: {
+      email: faker.internet.email({ provider: 'hackneyTEST.gov.uk' }),
+      name: faker.person.fullName(),
+      groups: Cypress.env('AUTHORISED_READONLY_GROUP'),
+    },
+  };
+
+  const user = users[userType];
+  if (!user) {
+    throw new Error(`No user data found for user type "${userType}"`);
+  }
+  const secret = 'aDummySecret';
+
+  cy.task('generateToken', { user, secret }).then((token) => {
+    const cookieName = 'hackneyToken';
+    cy.getCookies().should('be.empty');
+    cy.setCookie(cookieName, token as string);
+    cy.getCookie(cookieName).should('have.property', 'value', token);
```

Comment:
I don't think this assertion can fail unless cypress were to drop support for setCookie :thinking: .
Or is this a case of cypress not supporting setCookie on some browsers, but not others?

Reply:
Haven't given this a lot of thought aside from just checking the cookie set exists in the browser environment? Or am I missing something.

Reply:
It's probably fine. My comment was more so a question of whether there's any circumstance for when line 100 could fail?

As this reads somewhat like:

```cs
var token = "some string"
string myCookie = token;
myCookie.Should().Be(myCookie);
```

In essence, the assertion tests Cypress's in-built internal functionality. So I was wondering whether there are any cases when `cy.setCookie` is expected to fail to set the cookie?